core/src/main/java/com/linecorp/armeria/common/util/AbstractThreadFactory.java

```diff
@@ -0,0 +1,103 @@
+        delegate = new ThreadFactoryImpl(requireNonNull(threadNamePrefix, "threadNamePrefix"));
+    }
+
+    protected AbstractThreadFactory(String threadNamePrefix, boolean daemon) {
+        delegate = new ThreadFactoryImpl(requireNonNull(threadNamePrefix, "threadNamePrefix"),
+                                              daemon);
+    }
+
+    protected AbstractThreadFactory(String threadNamePrefix, int priority) {
+        delegate = new ThreadFactoryImpl(requireNonNull(threadNamePrefix, "threadNamePrefix"),
+                                              priority);
+    }
+
+    protected AbstractThreadFactory(String threadNamePrefix, boolean daemon, int priority) {
+        delegate = new ThreadFactoryImpl(requireNonNull(threadNamePrefix, "threadNamePrefix"),
+                                              daemon, priority);
+    }
+
+    protected AbstractThreadFactory(String threadNamePrefix, boolean daemon, int priority,
+                                    @Nullable ThreadGroup threadGroup) {
+        delegate = new ThreadFactoryImpl(requireNonNull(threadNamePrefix, "threadNamePrefix"),
+                                              daemon, priority, threadGroup);
+    }
+
+    protected AbstractThreadFactory(String threadNamePrefix, boolean daemon, int priority,
+                                 @Nullable ThreadGroup threadGroup,
+                                 Function<? super Runnable, ? extends Runnable> taskFunction) {
+        delegate = new ThreadFactoryImpl(requireNonNull(threadNamePrefix, "threadNamePrefix"),
+                                              daemon, priority, threadGroup);
+        this.taskFunction = requireNonNull(taskFunction, "taskFunction");
+    }
+
+    @Override
+    public final Thread newThread(Runnable r) {
+        return delegate.newThread(taskFunction.apply(r));
+    }
+
+    abstract Thread newThread(@Nullable ThreadGroup threadGroup, Runnable r, String name);
+
+    private final class ThreadFactoryImpl extends DefaultThreadFactory {
+        ThreadFactoryImpl(String threadNamePrefix) {
+            super(threadNamePrefix);
+        }
+
+        ThreadFactoryImpl(String threadNamePrefix, boolean daemon) {
+            super(threadNamePrefix, daemon);
+        }
+
+        ThreadFactoryImpl(String threadNamePrefix, int priority) {
+            super(threadNamePrefix, priority);
+        }
+
+        ThreadFactoryImpl(String threadNamePrefix, boolean daemon, int priority) {
+            super(threadNamePrefix, daemon, priority);
+        }
+
+        ThreadFactoryImpl(String threadNamePrefix, boolean daemon, int priority,
+                                   @Nullable ThreadGroup threadGroup) {
+            super(threadNamePrefix, daemon, priority, threadGroup);
+        }
+
+        @Override
+        protected Thread newThread(Runnable r, String name) {
+            return AbstractThreadFactory.this.newThread(threadGroup, taskFunction.apply(r), name);
```

Comment:
This makes the `taskFunction` apply twice.

Reply:
Removed it. Thank you.